### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by Evolution0** [Evolution0/bandcamp-dl#85]
Currently trying to work out a neat and manageable documentation method, personally I favor Function Annotation + Docstrings in ReStructuredText.

I'm currently setting the code up as such in my PyCharm project along with the pytests.

Should the PyCharm project files be included in future commits?

**Comment by iheanyi** [comment]
No idea with regards to PyCharm. Personally, I use vim.

**Comment by Evolution0** [comment]
So far I have added Function Annotation, Return Types, and Docstrings where possible, There are a few more that could be added like on some of the class inits but other than that its mostly complete.

Whats to be done next is:
1. Give the readme a once over.
   * Verify all information is correct and remove anything that isn't or is irrelevant.
   * Details and Options can be merged into one section, that section can then be generated automatically with Sphinx from the docstring in bandcamp_dl.py.
   * Filename template can be explained in the Wiki along with the rest of the features.
2. Update Wiki.
   * Long overdue, a nice explanation of the features and options in plain english would due nicely.
   * Provide examples for each command.
   * Revamp the "For Dummies" section.

Doing so has made me realize that some things could be changed, parse no longer really parses anything (or never did), it just takes the values out of the dict its given and cherry picks what values we want to pass as the album dict. All parsing is done by `generate_album_json()` and subsequently `BandcampJSON`.

A rename might be due (possibly build_album?) once renamed a proper function for individual track downloads can be made as I'm not a fan of the "It kinda just works" aspect of it currently.

Track downloads only work because most track links are part of an album and the required metadata can still be retrieved.

I'm still not 100% sure what would happen if a track url was provided in which the track is *not* part of an album. This has apparently happened before and an issue was made but it was old and no links were provided in which to test with and I can't seem to find any in which to do so.

**Comment by iheanyi** [comment]
Yeah, some of these edge cases are tricky to debug, tbh. Hmmmm.  I wish there was an easy way to find an example that breaks. Perhaps we should make an ote in the README if they are having issues to include example links so we can debug it/reproduce it?

**Comment by Evolution0** [comment]
In the 0.0.7 branch I have added the .github directory which contains an issue template so users will know exactly what information needs to be provided upon creating an issue from now on.

The contributing and pull request template will also go there.

**Comment by Evolution0** [comment]
Wiki has been updated.
